packages/rust/server/src/models/group.rs

```diff
@@ -0,0 +1,224 @@
+        match value {
+            0 => Ok(Role::Admin),
+            1 => Ok(Role::Member),
+            _ => Err(()),
+        }
+    }
+}
+
+#[derive(sqlx::FromRow)]
+pub struct UserIDs {
+    pub user_ids: Vec<String>,
+}
+
+#[derive(sqlx::FromRow)]
+pub struct GroupIDs {
+    pub group_ids: Vec<UserGroup>,
+}
+
+impl From<&PgRow> for Group {
+    fn from(row: &PgRow) -> Self {
+        Group {
+            id: row.get("id"),
+            uuid: row.get("uuid"),
+            group_name: row.get("group_name"),
+            created: row.get("created"),
+            members: Vec::new(),
+        }
+    }
+}
+
+impl Group {
+    pub async fn delete_user_groups(db: &DbPool, user_id: i32) -> Result<(), sqlx::Error> {
+        sqlx::query("DELETE FROM user_group WHERE user_id = $1")
+            .bind(user_id)
+            .execute(db)
+            .await?;
+        Ok(())
+    }
+
+    pub async fn role(&self, db: &DbPool, user_id: i32) -> Result<Role, sqlx::Error> {
+        let row = sqlx::query("SELECT * FROM user_group WHERE group_id = $1 AND user_id = $2;")
+            .bind(self.id)
+            .bind(user_id)
+            .fetch_one(db)
+            .await?;
+        let role_id: i32 = row.get("role_id");
+        let role_enum = match Role::try_from(role_id as u8) {
+            Ok(role_enum) => role_enum,
+            Err(_) => return Err(sqlx::Error::RowNotFound),
+        };
+        Ok(role_enum)
+    }
+
+    pub async fn members(&self, db: &DbPool) -> Result<Vec<User>, sqlx::Error> {
+        let row = sqlx::query("SELECT * FROM user_group WHERE group_id = $1 ")
+            .bind(self.id)
+            .fetch_all(db)
+            .await?;
+        let mut users_in_group: Vec<User> = vec![];
+        for i in row {
+            let user_to_be_added = User::from_id(db, i.get("id")).await?;
+            users_in_group.push(user_to_be_added);
+        }
+        Ok(users_in_group)
+    }
+
+    pub async fn from_uuid(db: &DbPool, uuid: &Uuid) -> Result<Group, sqlx::Error> {
+        Ok(sqlx::query("SELECT * FROM \"group\" WHERE uuid = $1")
+            .bind(uuid)
+            .fetch_one(db)
+            .await?
+            .borrow()
+            .into())
+    }
+
+    pub async fn from_id(db: &DbPool, id: i32) -> Result<Group, sqlx::Error> {
+        Ok(sqlx::query("SELECT * FROM \"group\" WHERE id = $1")
+            .bind(id)
+            .fetch_one(db)
+            .await?
+            .borrow()
+            .into())
+    }
+
+    pub async fn create(&mut self, db: &DbPool, user_id: i32) -> Result<(), sqlx::Error> {
+        let mut tx = db.begin().await?;
+        let row =
+            sqlx::query("INSERT INTO \"group\" (uuid, group_name) VALUES ($1, $2) RETURNING *;")
+                .bind(&self.uuid)
+                .bind(&self.group_name)
+                .fetch_one(&mut tx)
+                .await?;
+        self.id = row.get("id");
+        self.created = row.get("created");
+        sqlx::query(
```

Comment:
i assume there is a method to add a user to a group, you should probably call that here instead of remaking it

Reply:
fixed, changed create_group